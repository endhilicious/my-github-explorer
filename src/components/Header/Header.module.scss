.header {
  padding: 1rem;
  text-align: center;
  h1 {
    margin: 0;
    font-size: 2rem;
  }
}
