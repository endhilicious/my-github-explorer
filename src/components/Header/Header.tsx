import React from 'react';

import styles from './Header.module.scss';

const Header: React.FC = () => {
  return (
    <header className={styles.header}>
      <h1 data-testid="header-text">G ithub Re positories Explorer</h1>
    </header>
  );
};

export default Header;
