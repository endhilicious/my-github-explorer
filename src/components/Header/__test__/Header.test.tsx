import React from 'react';
import { render } from '@testing-library/react';
import Header from '../Header';

test('Header menampilkan teks yang benar', () => {
  const { getByTestId } = render(<Header />);
  const headerElement = getByTestId('header-text');

  expect(headerElement.textContent).toBe('Github Repositories Explorer');
});

test('Header memiliki tampilan yang konsisten dengan snapshot', () => {
  const { asFragment } = render(<Header />);

  expect(asFragment()).toMatchSnapshot();
});
