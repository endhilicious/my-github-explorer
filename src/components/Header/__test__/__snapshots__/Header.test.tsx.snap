// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header memiliki tampilan yang konsisten dengan snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <h1
      data-testid="header-text"
    >
      Github Repositories Explorer
    </h1>
  </header>
</DocumentFragment>
`;
